subt/docker/robotika/ros_proxy_node.cc

```diff
@@ -621,45 +623,21 @@ void Controller::receiveZmq(const ros::TimerEvent&)
 /////////////////////////////////////////////////
 int main(int argc, char** argv)
 {
+  if (argc < 2 || std::strlen(argv[1]) == 0) {
```

Comment:
Are you sure the name is passed to ros_proxy_node? As far as I remember the name "autodetection" was used/needed??

Reply:
`x1.launch` has a parameter called `robot_name` and its default value is empty string. So when run through the launch file, we have always at least one parameter, only sometimes is empty. I have moved the detection of the robot name into the outer script and ensure the supplied `robot_name` is not empty.